docs/userguide/expectations.md

```diff
@@ -0,0 +1,78 @@
+# Concept of expectation and matcher 
+
+Validation of the code under test (the tested logic of procedure/function etc.) is performed by comparing the actual data against the expected data.
+To do that we use concept of expectation and a matcher to perform the check on the data.
+
+It's best to give an example to get an idea what is what
+```sql
+begin
+  ut.expect( 'the tested value' ).to_( equal('the expected value') );
+end;
+```
+
+Expectation is a set of the expected value(s), actual values(s) and the matcher(s) to run on those values.
+
+Matcher is defining the comparison operation to be performed on expected and actual values. 
+
+# List of currently build-in matchers
+- `match`
+- `equal`
+- `be_true`
+- `be_null`
+- `be_not_null`
+- `be_like`
+- `be_less_than`
+- `be_less_or_equal`
+- `be_greater_than`
+- `be_greater_or_equal`
+- `be_false`
+- `be_between`
+
+## match
+Allows regexp_like validations to be executed against the following datatypes:
+- `clob`
+- `varchar2`
+
+Usage:
+```sql
+  ut.expect( a_actual ).to_( match( a_pattern in varchar2, a_modifiers in varchar2 := null) )
+```
+
+Parameters `a_pattern` and `a_modifiers` represent a valid regexp pattern accepted by [Oracle regexp_like function](https://docs.oracle.com/cd/E11882_01/server.112/e41084/conditions007.htm#SQLRF00501)
```

Comment:
change link to http://docs.oracle.com/database/121/SQLRF/conditions007.htm#SQLRF52141